fractal-puppy/Engine
# Game Data Serialization

# Player Prefs

Static class that allows to store simple data: *int*, *float* and *const char**.
- There is a Setter and a Getter for each type of data. It is a key-value structure, for instance:
```cpp
PlayerPrefs::SetInt("health", 80)
int health = PlayerPrefs::GetInt("health); //health now is equal to 80
```

- You can check if a key is already saved. It returns true if found:
```cpp
bool playerHasHighscore = PlayerPrefs::HasKey("highscore")
```

- You can delete a given key:
```cpp
PlayerPrefs::DeleteKey("mana")
```
- You can delete all the information saved in PlayerPrefs:
```cpp
PlayerPrefs::DeleteAll();
```
This information is persistent across scenes!!

- For more complex information there are two methods, *LoadJson* and *SaveJson*. You can create your own *JSON* and load and store it with PlayerPrefs.

## Testing
You can use whatever script you prefer.
- For testing purposes use a Getter and a Setter and check if they work properly.
- Close and open the scene and check that the information is saved.
- Use the HasKey and DeleteKey methods with a created key and check that it exists and it is correctly deleted.
- Finally use DeleteAll and check that all the keys created no longer exist.
- Check that LoadJson and SaveJson work creating a small JSON and storing and loading it.
